extensions/amp-list/0.1/amp-list.js

```diff
@@ -780,19 +756,22 @@ export class AmpList extends AMP.BaseElement {
    * @private
    */
   toggleLoadMoreLoading_(state) {
-    if (this.loadMoreLoadingElement_) {
-      this.mutateElement(() => {
-        if (state) {
-          this.loadMoreButton_.classList.toggle('amp-visible', false);
-        }
+    this.mutateElement(() => {
+      // If it's loading, then it's no longer failed
+      if (this.loadMoreFailedElement_) {
+        this.loadMoreFailedElement_.classList.toggle('amp-visible', false);
+      }
+      if (this.loadMoreLoadingElement_) {
+        this.loadMoreButton_.classList.toggle('amp-visible', !state);
         this.loadMoreLoadingElement_.classList.toggle('amp-visible', state);
-      });
-    } else if (this.loadMoreButton_) {
-      this.mutateElement(() => {
+      } else if (this.loadMoreButton_) {
```

Comment:
Should this be an else-if? It looks like both `this.loadMoreLoadingElement_` and `this.loadMoreButton_` are defined together, so I think this else-if branch will never run if both are present on the element.

https://github.com/ampproject/amphtml/blob/d0aa98a2ae233cfdc944a856af41c04a632ef3e7/extensions/amp-list/0.1/amp-list.js#L183-L185

Reply:
This is intended. Basically if we have a `loading` element, we should show / hide that. Otherwise, the current code toggles a CSS class on the `load-more-button` element and shows a loader which was created (`this.loadMoreLoadingOverlay_`). This logic may change in the future after we have UX defaults, depending on what those defaults look like.